doc/source/analysis.rst

```diff
@@ -1,5 +1,400 @@
+
+If you prefer a vanilla edge list, you can use :meth:`Graph.get_edge_list`.
+
+Incidence
+++++++++++++++++++++++++++++++
+To get the vertices at the two ends of an edge, use :attr:`Edge.source` and :attr:`Edge.target`:
+
+>>> e = g.es[0]
+>>> v1, v2 = e.source, e.target
+
+Vice versa, to get the edge if from the source and target vertices, you can use :meth:`Graph.get_eid` or, for multiple pairs of source/targets,
+:meth:`Graph.get_eids`. The boolean version, asking whether two vertices are directly connected, is :meth:`Graph.are_connected`.
+
+To get the edges incident on a vertex, you can use :meth:`Vertex.incident`, :meth:`Vertex.out_edges` and
+:meth:`Vertex.in_edges`. The three are equivalent on undirected graphs but not directed ones, of course:
+
+>>> v = g.vs[0]
+>>> edges = v.incident()
+
+The :meth:`Graph.adjecent` and :meth:`Graph.incident` functions fulfill the same purpose with a slightly different syntax based on vertex ids:
+
+>>> edges = g.incident(0)
+
+To get the full adjacency/incidence list representation of the graph, use :meth:`Graph.get_adjlist`, :meth:`Graph.g.get_inclist()` or, for a bipartite graph, :meth:`Graph.get_incidence`.
+
+Neighborhood
++++++++++++++
+To compute the neighbors, successors, and predecessors, the methods :meth:`Graph.neighbors`, :meth:`Graph.successors` and
+:meth:`Graph.predecessors` are available. The three give the same answer in undirected graphs and have a similar dual syntax:
+
+>>> neis = g.vs[0].neighbors()
+>>> neis = g.neighbors(0)
+
+To get the list of vertives within a certain distance of one or more initial nodes, you can use :meth:`Graph.neighborhood`:
+
+>>> g.neighborhood([0, 1], order=2)
+
+and to find the neighborhood size, there is :meth:`Graph.neighborhood_size`.
+
+Degrees
++++++++
+To compute the degree, in-degree, or out-degree of a node, use :meth:`Vertex.degree`, :meth:`Vertex.indegree`, and :meth:`Vertex.outdegree`:
+
+>>> deg = g.vs[0].degree()
+>>> deg = g.degree(0)
+
+To compute the max degree in a list of vertices, use :meth:`Graph.maxdegree`.
+
+:meth:`Graph.knn` computes the average degree of the neighbors.
+
+Adding and removing vertices and edges
+++++++++++++++++++++++++++++++++++++++
+
+To add nodes to a graph, use :meth:`Graph.add_vertex` and :meth:`Graph.add_vertices`:
+
+>>> g.add_vertex()
+>>> g.add_vertices(5)
+
+This changes the graph `g` in place. You can specify the name of the vertices if you wish.
+
+To remove nodes, use :meth:`Graph.delete_vertices`:
+
+>>> g.delete_vertices(None)  # remove all vertices
+>>> g.delete_vertices([1, 2])
+
+Again, you can specify the names or the actual :class:`Vertex` objects instead.
+
+To add edges, use :meth:`Graph.add_edge` and :meth:`Graph.add_edges`:
+
+>>> g.add_edge(0, 2)
+>>> g.add_edges([(0, 2), (1, 3)])
+
+To remove edges, use :meth:`Graph.delete_edges`:
+
+>>> g.delete_edges(None)   # remove all edges
```

Comment:
Same here - this is an unintended consequence of how we use `igraphmodule_PyObject_to_es_t()` to convert `None` to "all edges". Calling `delete_edges()` with `None` is discouraged and in the future it will probably throw an error.

Reply:
yep removed, will do a separate PR on that